zebra/zebra_routemap.c

```diff
@@ -1549,6 +1585,59 @@ static const struct route_map_rule_cmd route_set_src_cmd = {
 	route_set_src_free,
 };
 
+static enum route_map_cmd_result_t
+route_set_kernel_bypass(void *rule, const struct prefix *prefix,
+				      void *object)
+{
+	struct nh_rmap_obj *nh_data;
+	char ipv6_str[INET6_ADDRSTRLEN];
+	nh_data = (struct nh_rmap_obj *)object;
+	SET_FLAG(nh_data->nexthop->flags,NEXTHOP_FLAG_KERNEL_BYPASS);
+	if (unlikely(IS_ZEBRA_DEBUG_KERNEL))
+	{
+		switch (nh_data->nexthop->type) {
+			case NEXTHOP_TYPE_IPV4_IFINDEX:
+			case NEXTHOP_TYPE_IPV4:
+				zlog_debug("zebra-route-map:set kernel-bypass flag on nh,src: %pI4 gate: %pI4, nh flags:0x%x",
+					&nh_data->nexthop->src.ipv4,
+					&nh_data->nexthop->gate.ipv4,
+					nh_data->nexthop->flags
+					);
+				break;
+			case NEXTHOP_TYPE_IPV6_IFINDEX:
+			case NEXTHOP_TYPE_IPV6:
+				inet_ntop(AF_INET6, &nh_data->nexthop->gate.ipv6, ipv6_str, INET6_ADDRSTRLEN);
+				zlog_debug("zebra-route-map:set kernel-bypass flag on nh, gate: %pI6, ,nh flags:0x%x",
+					&nh_data->nexthop->gate.ipv6,
+					nh_data->nexthop->flags);
+				break;
+			case NEXTHOP_TYPE_BLACKHOLE:
+			case NEXTHOP_TYPE_IFINDEX:
+				break;
+	}
+
+	}
+	
+	return RMAP_OKAY;
+}
+static void *route_set_kernel_bypass_compile(const char *arg)
+{
+	return (void*) 1;
```

Comment:
`return XSTRDUP(MTYPE_ROUTE_MAP_COMPILED, arg);` - to be consistent with others.

Reply:
I learnt this return way from route-map "atomic-aggregate". Actually when I use `XSTRDUP`, it return NULL. Because this route map will be like "set kernel-bypass", which means "arg" of `route_set_kernel_bypass_compile` is NULL.